src/domainobjectfactories/cash_flow_factory.py

```diff
@@ -0,0 +1,134 @@
+from datetime import datetime, timezone
+import random
+
+from domainobjectfactories.creatable import Creatable
+
+
+class CashFlowFactory(Creatable):
+    """ Class to create cash flows. Create method will create
+        a set amount of cash flows."""
+    PAYMENT_STATUSES = ['Actual', 'Contractual']
+
+    def create(self, record_count, start_id):
+        """ Create a set number of cash flows
+
+        Parameters
+        ----------
+        record_count : int
+            Number of cash flows to create
+        start_id : int
+            Starting id to create from
+
+        Returns
+        -------
+        List
+            Containing 'record_count' cash flows
+        """
+
+        records = []
+
+        for _ in range(start_id, start_id+record_count):
+            records.append(self.__create_record())
+
+        return records
+
+    def __create_record(self):
+        """ Create a single cash flow
+
+        Returns
+        -------
+        dict
+            A single cash flow object
+        """
+
+        record = {
+            'account_id': self.__get_account_id(),
+            'corporate_action_id': self.__create_corporate_action_id(),
+            'quantity': self.__get_quantity(),
+            'currency': self.__get_currency(),
+            'payment_status': self.__get_payment_status(),
+            'payment_type': self.__get_payment_type(),
+            'payment_date': self.__get_payment_date()
+        }
+
+        for key, value in self.create_dummy_field_generator():
+            record[key] = value
+
+        return record
+
+    def __get_account_id(self):
+        """ Return a account id from an account persisted in the database where
+        account type is 'Client' or 'Firm'
+
+        Returns
+        -------
+        String
+            account id from the local database
+        """
+        account = self.get_random_record_with_valid_attribute(
+            'accounts', 'account_type', ['Client', 'Firm']
+        )
+        account_id = account['account_id']
+        return account_id
+
+    def __create_corporate_action_id(self):
+        """ Return corporate action id dummy string
+
+        Returns
+        -------
+        String
+            randomly generated 10 character dummy string
+        """
+        return self.create_random_string(10)
+
+    def __get_quantity(self):
+        """ Return quantity of cash in this cashflow
+            Returns
+            -------
+            Double
```

Comment:
Nitpick but this is Java-speak rather than Python -0 the return type would be `float` not `Double`